src/bson.c

```diff
@@ -536,14 +619,33 @@ static bool php_phongo_bson_visit_symbol(const bson_iter_t* iter, const char* ke
 	return false;
 } /* }}} */
 
+static void php_phongo_bson_new_javascript_from_javascript_and_scope(int init, zval* object, const char* code, size_t code_len, const bson_t* scope TSRMLS_DC) /* {{{ */
```

Comment:
Currently, `php_phongo_new_javascript_from_javascript_and_scope` and `php_phongo_new_javascript_from_javascript` are only called with `init=1` from `bson.c`.

The `init` argument here dates back to https://github.com/mongodb/mongo-php-driver/commit/ab8d22c7ffcbace3d383a951ad0c40fbe15ac025, when `Javascript::__construct()` used to rely on this method. Since the PHP constructor initializes the object before the extension's method is even called, we needed a flag to toggle whether we should initialize in this function. That was ultimately removed in https://github.com/mongodb/mongo-php-driver/commit/41081dae61c56003a510da42484d617b5cfaad64 when we introduced serialization and `var_export` support for the Javascript BSON type, which means `init` no longer serves a purpose.

We should be able to remove it entirely, which will make this function consistent with the other BSON types.